core/src/rpc.rs

```diff
@@ -413,6 +413,30 @@ impl JsonRpcRequestProcessor {
             .ok()
             .unwrap_or(None))
     }
+
+    pub fn get_transaction_status_history(
+        &self,
+        signature: Signature,
+    ) -> RpcResponse<Option<RpcTransactionStatus>> {
+        let root = self.bank_forks.read().unwrap().root();
+        self.blockstore
+            .get_transaction_status(signature, root)
```

Comment:
Oh, this new endpoint will only return statuses that have been rooted? This would make clients pretty complex if they wanted to use non-rooted statuses. They would have to use a combination of APIs to track the status. I agree with @mvines that we should combine endpoints and also allow querying for non-rooted slots

Reply:
No, this endpoint definitely returns non-rooted statuses as well. We just need to know the current root to determine if the associated slot is dead or just not rooted yet.

Reply:
Ah whoops, my mistake. I see that now